src/main/java/org/gradle/profiler/Profiler.java

```diff
@@ -18,15 +18,25 @@
 import java.io.File;
 import java.util.function.Consumer;
 
-public class Profiler {
+public abstract class Profiler {
 
     public static final Profiler NONE = new Profiler() {
+        @Override
+        public boolean requiresGradle() {
+            return false;
+        }
+
         @Override
         public String toString() {
             return "none";
         }
     };
 
+    /**
+     * Whether this profiler supports only Gradle builds.
+     */
+    public abstract boolean requiresGradle();
```

Comment:
🤔 It's a bit unfortunate we have this `boolean requiresGradle()`, maybe we could have some more generic method.

Reply:
Yes, 100%. Again, let's discuss how we want to model this relationship.

Reply:
Essentially, there are profilers that can run via Java agent, and profilers that can run as an external process, and profilers that support both.

And then there are build tools that support supplying an agent (Gradle, Maven), build tools that support a long running daemon (Gradle only), and build tools that support neither (Buck and Bazel might be these, at least as a start).

I think this can be modeled nicely, and will make the compatibility between tool and profiler be obvious.

Reply:
https://github.com/gradle/gradle-profiler/pull/576